apps/salesforce-commerce-cloud/src/components/dialog/SearchPicker.tsx

```diff
@@ -0,0 +1,150 @@
+  const { selectMultiple, fieldType, fieldValue, currentData } = sdk.parameters
+    .invocation as DialogInvocationParameters;
+
+  const [query, setQuery] = useState<string>('');
+  const [queryIsFetching, setQueryIsFetching] = useState<boolean>(false);
+  const [selected, setSelected] = useState<string | string[] | undefined>(fieldValue);
+  const [selectedData, setSelectedItemsInfo] = useState<any[]>(currentData || []);
+  const [searchResults, setSearchResults] = useState<any[]>([]);
+
+  const onQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const onSave = () => {
+    sdk.close(selected);
+  };
+
+  const findSearchResultData = (id: string) => {
+    return searchResults.find((item) => {
+      return fieldType === 'product' ? item.id === id : `${item.catalogId}:${item.id}` === id;
+    });
+  };
+
+  const onItemSelect = (id: string) => {
+    if (selectMultiple) {
+      // Multivalue
+      if (!selected?.length) {
+        // Empty array, Initial Value
+        setSelected([id]);
+        setSelectedItemsInfo([findSearchResultData(id)]);
+      } else {
+        // Array exists, modify it
+
+        const updateSelected = [...selected];
+        let updateSelectedData = [];
+        const includedIndex = updateSelected.findIndex((item) => item === id);
+        if (includedIndex > -1) {
+          // Item exists in array, unset it
+          updateSelected.splice(includedIndex, 1);
+          updateSelectedData = selectedData.filter((item) => {
+            return fieldType === 'product' ? item.id === id : `${item.catalogId}:${item.id}` === id;
+          });
+        } else {
+          console.log('Item not found.');
+          updateSelected.push(id);
+          updateSelectedData = [...selectedData, findSearchResultData(id)];
+        }
+
+        setSelected(updateSelected);
+        setSelectedItemsInfo(updateSelectedData);
+      }
+    } else {
+      // Single value
+      if (selected === id) {
+        setSelected('');
+        setSelectedItemsInfo([]);
+      } else {
+        setSelected(id);
+        setSelectedItemsInfo([findSearchResultData(id)]);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const timeOutId = setTimeout(() => {
+      const client = new SfccClient(installParameters);
+      const fetchResults =
+        fieldType === 'product' ? client.searchProducts : client.searchCategories;
+      if (query.length >= 3) {
+        setQueryIsFetching(true);
+        fetchResults(query)
+          .then((results: any[]) => setSearchResults(results))
+          .finally(() => setQueryIsFetching(false));
+      } else {
+        setQueryIsFetching(true);
+        fetchResults()
+          .then((results: any[]) => setSearchResults(results))
+          .finally(() => setQueryIsFetching(false));
+      }
+    }, 500);
+    return () => clearTimeout(timeOutId);
+  }, [fieldType, query, installParameters]);
```

Comment:
I saw we imported reactQuery at the top level, why are we not using the queryClient here?